### ISSUE TITLE
Focal Loss

### ISSUE THREAD

**Issue by StevenZhangzhexu** [drprojects/superpoint_transformer#56]
Hi @rjanvier , I saw there was an implementation src/loss/focal.py of focal loss and I would like to apply it as loss function during traning on my dataset (imbalanced). Could you please advise how to do so? Thanks

**Comment by drprojects** [comment]
Hi, you can try moving from CrossEntropy to FocalLoss by changing [these lines](https://github.com/drprojects/superpoint_transformer/blob/master/configs/model/default.yaml#L28-L29) to:

```yaml
criterion:
     _target_: src.loss.FocalLoss
```

In my experiments, this did not change much for the benchmarked datasets. The code for the Focal loss and Lovasz lossare remnant of past experiments, I cannot guarantee that they will work well with the multi-stage loss formulation (triggered by the `multi_stage_loss_lambdas` parameter). If it causes any issue, you can turn it off by setting `multi_stage_loss_lambdas: null` in the corresponding config.
